### PR TITLE
Refactor IconThemeData.fallback

### PR DESCRIPTION
**Pull request by abarth**
Now IconThemeData.fallback is a factory constructor and IconThemeData.of() does
the work of computing the fallback for its clients.

Also, add tests for ImageIcon and ListItems.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/material/icon_theme.dart [hunk 1]
```diff
@@ -56,8 +56,13 @@ class IconTheme extends InheritedWidget {
   /// IconThemeData theme = IconTheme.of(context);
   /// ```
   static IconThemeData of(BuildContext context) {
-    IconTheme result = context.inheritFromWidgetOfExactType(IconTheme);
-    return result?.data ?? Theme.of(context).iconTheme;
+    IconThemeData themeData = _getInheritedIconThemData(context);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/icon_theme.dart]: can you call this variable iconThemeData? I briefly got confused and thought it was a ThemeData.

File: packages/flutter/lib/src/material/icon_theme.dart [hunk 2]
```diff
@@ -56,8 +56,13 @@ class IconTheme extends InheritedWidget {
...
+    return themeData.isConcrete ? themeData : const IconThemeData.fallback().merge(themeData);
+  }
+
+  static IconThemeData _getInheritedIconThemData(BuildContext context) {
+    IconTheme theme = context.inheritFromWidgetOfExactType(IconTheme);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/icon_theme.dart]: similar here but iconTheme

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/icon_theme.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/material/icon_theme.dart]: Sure.

**Pull request merged**